File: longest-common-prefix.py
# Question Link - https://leetcode.com/problems/longest-common-prefix/
#Difficulty - Medium

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        commonPrefix = ""
        for i in range(len(strs[0])):
            for word in strs:
                if i >= len(word) or word[i] != strs[0][i]:
                    return commonPrefix
            commonPrefix += strs[0][i]

        return commonPrefix

File: longest-common-prefix.js
// Question link - https://leetcode.com/problems/longest-common-prefix/
// difficulty - Medium

/**
 * @param {string[]} strs
 * @return {string}
 */
var longestCommonPrefix = function(strs) {
    
    let common = "";
    let smIndex=0;
    for(let i=0; i<strs[smIndex].length; i++) {
        for(let j=0; j<strs.length; j++) {
                if (strs[smIndex][i] !== strs[j][i]) {
                    return common;
                }
        }
        common += strs[smIndex][i];
    }
    
    return common;
    
};